tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Services/ProfileServiceTests.cs

```diff
@@ -189,8 +189,25 @@ public static async Task GetProfileDataAsync_GivenAnApplicationUserExistsWithOrg
             actual.Should().BeEquivalentTo(expectedClaimTypes);
         }
 
-        [TestCase("someone@email.com", Subject, Email, EmailVerified, ApplicationClaimTypes.PrimaryOrganisationId, ApplicationClaimTypes.PrimaryOrganisationName, ApplicationClaimTypes.OrganisationFunction, ApplicationClaimTypes.Ordering, ApplicationClaimTypes.Organisation)]
-        [TestCase("", Subject, ApplicationClaimTypes.PrimaryOrganisationId, ApplicationClaimTypes.PrimaryOrganisationName, ApplicationClaimTypes.OrganisationFunction, ApplicationClaimTypes.Ordering, ApplicationClaimTypes.Organisation)]
+        [TestCase(
+            "someone@email.com",
+            Subject,
+            Email,
+            EmailVerified,
+            ApplicationClaimTypes.PrimaryOrganisationId,
+            ApplicationClaimTypes.PrimaryOrganisationName,
+            ApplicationClaimTypes.OrganisationFunction,
+            ApplicationClaimTypes.Ordering,
+            ApplicationClaimTypes.Organisation)]
+
+        [TestCase(
+            "",
+            Subject,
+            ApplicationClaimTypes.PrimaryOrganisationId,
+            ApplicationClaimTypes.PrimaryOrganisationName,
+            ApplicationClaimTypes.OrganisationFunction,
+            ApplicationClaimTypes.Ordering,
+            ApplicationClaimTypes.Organisation)]
```

Comment:
These test cases are a tad unwieldy aren't they? 😃 I'm wonder if it's worth using a test case source... Do you find something like this any more readable (assuming `GetSomeTestCaseData` is better named)?

```c#
private static IEnumerable<ITestCaseData> GetSomeTestCaseData()
{
    yield return new TestCaseData("someone@email.com", new[]
    {
        Subject,
        Email,
        EmailVerified,
        ApplicationClaimTypes.PrimaryOrganisationId,
        ApplicationClaimTypes.PrimaryOrganisationName,
        ApplicationClaimTypes.OrganisationFunction,
        ApplicationClaimTypes.Ordering,
        ApplicationClaimTypes.Organisation,
    });

    yield return new TestCaseData(string.Empty, new[]
    {
        Subject,
        ApplicationClaimTypes.PrimaryOrganisationId,
        ApplicationClaimTypes.PrimaryOrganisationName,
        ApplicationClaimTypes.OrganisationFunction,
        ApplicationClaimTypes.Ordering,
        ApplicationClaimTypes.Organisation,
    });
}

[TestCaseSource(nameof(GetSomeTestCaseData))]
public static async Task GetProfileDataAsync_GivenApplicationUserWithEmail_ReturnExpectedClaimList(
```

Reply:
the only issue is that each test case is unique, so we'd need a testcasedata for each test as such, which looking through would mean we'd need 3 of these, which wouldn't save any space really.

but i'm not against it, just depends which one you would prefer

Reply:
Not too fussed really – there's pros and cons to each so feel free to leave as is.